src/dune_rules/workspace.ml

```diff
@@ -279,20 +279,36 @@ module Context = struct
       { base : Common.t
       ; lock : Path.Source.t option
       ; version_preference : Dune_pkg.Version_preference.t option
-      ; solver_env : Dune_pkg.Solver_env.t option
+      ; override_solver_sys_vars : Dune_pkg.Solver_env.Variable.Sys.Bindings.t option
+      ; default_repositories : Dune_pkg.Pkg_workspace.Repository.Name.t list
       }
 
-    let to_dyn { base; lock; version_preference; solver_env } =
+    let to_dyn
+      { base; lock; version_preference; override_solver_sys_vars; default_repositories }
+      =
       Dyn.record
         [ "base", Common.to_dyn base
         ; "lock", Dyn.(option Path.Source.to_dyn) lock
         ; ( "version_preference"
           , Dyn.option Dune_pkg.Version_preference.to_dyn version_preference )
-        ; "solver_env", Dyn.option Dune_pkg.Solver_env.to_dyn solver_env
+        ; ( "override_solver_sys_vars"
+          , Dyn.option
+              Dune_pkg.Solver_env.Variable.Sys.Bindings.to_dyn
+              override_solver_sys_vars )
+        ; ( "default_repositories"
+          , Dyn.list Dune_pkg.Pkg_workspace.Repository.Name.to_dyn default_repositories )
         ]
     ;;
 
     let decode =
+      let default_repositories_of_ordered_set ordered_set =
+        Dune_lang.Ordered_set_lang.eval
+          ordered_set
+          ~parse:(fun ~loc string ->
+            Dune_pkg.Pkg_workspace.Repository.Name.parse_string_exn (loc, string))
+          ~eq:Dune_pkg.Pkg_workspace.Repository.Name.equal
+          ~standard:[ Dune_pkg.Pkg_workspace.Repository.Name.of_string "default" ]
```

Comment:
While at it, it would be good to unify `:standard` with the `default` repositories, to avoid the case that `:standard` and `` (no stanza) end up having different defaults.

(One could also refactor this where the name would be taken from the predefined repository, but I don't think hoisting more work onto this PR is sensible, better get iterative improvements).

Reply:
> While at it, it would be good to unify `:standard` with the `default` repositories

@Leonidas-from-XIV mind if I leave this to you? I don't feel like I have context for doing that at the moment.

Reply:
Yeah, sure, it's not urgent at all and not related to this PR really.